### PR TITLE
Zip317 add propose command

### PR DESCRIPTION
**Pull request by Oscar-Pepper**
- Adds propose and quicksend commands
- Adds unimplemented do_propose and do_send_proposal lightclient methods

### CODE REVIEW

**Review by zancas — DISMISSED** [review]

**Comment by fluidvanadium** [comment]
this looks good. the only conflict is the changelog.

File: zingolib/src/commands.rs [hunk 1]
```diff
@@ -824,16 +820,69 @@ impl Command for SendCommand {
                 )
             }
         };
-        for send in &send_inputs {
-            let address = &send.0;
-            let memo = &send.2;
-            if memo.is_some() && is_transparent_address(address, &lightclient.config.chain) {
+        if let Err(e) = utils::check_memo_compatibility(&send_inputs, &lightclient.config().chain) {
```
Comment (zancas) [inline zingolib/src/commands.rs:823]: Please helperize wet code.

File: zingolib/src/commands/utils.rs [hunk 2]
```diff
@@ -76,6 +80,22 @@ pub(super) fn parse_send_args(
     Ok(send_args)
 }
 
+/// Checks send inputs do not contain memo's to transparent addresses.
```
Comment (zancas) [inline zingolib/src/commands/utils.rs]: If possible add a doc-test, if that's not possible a unit-test, if that's not possible an explicit TODO.

File: zingolib/src/lightclient/send.rs [hunk 3]
```diff
@@ -53,6 +60,23 @@ impl LightClient {
             .collect()
     }
 
+    /// Unstable function to expose the zip317 interface for development
```
Comment (zancas) [inline zingolib/src/lightclient/send.rs:63]: doc-test, or unit-test, or TODO

**Review by zancas — CHANGES_REQUESTED** [review]

File: zingolib/src/commands.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply zingolib/src/commands.rs:823]: I did helperise into check_memo_compatibility as requested ... this error handling looks dry to me. .. can you provide a suggestion?

File: zingolib/src/commands/utils.rs (hunk 2 above)
Comment (Oscar-Pepper) [reply zingolib/src/commands/utils.rs]: will do

File: zingolib/src/lightclient/send.rs (hunk 3 above)
Comment (Oscar-Pepper) [reply zingolib/src/lightclient/send.rs:63]: i can add a TODO but this function is clearly still under a lot of development anyways

File: zingolib/src/commands/utils.rs (hunk 2 above)
Comment (Oscar-Pepper) [reply zingolib/src/commands/utils.rs]: oh this is not pub, i will add a unit test

File: zingolib/src/lightclient/send.rs [hunk 4]
```diff
@@ -53,7 +60,26 @@ impl LightClient {
...
+        use crate::test_framework::mocks::ProposalBuilder;
+
+        Ok(ProposalBuilder::default().build())
+    }
+
+    /// Unstable function to expose the zip317 interface for development
+    // TODO: add correct functionality and doc comments / tests
+    #[cfg(feature = "zip317")]
+    pub async fn do_send_proposal(&self) -> Result<Vec<TxId>, String> {
+        Ok(vec![TxId::from_bytes([0u8; 32])])
```
Comment (fluidvanadium) [inline zingolib/src/lightclient/send.rs:79]: i created a helper function in https://github.com/zingolabs/zingolib/pull/942/files called default_txid

**Review by fluidvanadium — DISMISSED** [review]
Looks like comments have been addressed and PR is ready

File: zingolib/src/lightclient/send.rs (hunk 3 above)
Comment (zancas) [reply zingolib/src/lightclient/send.rs:63]: OK.  This fulfills my requests.

**Review by zancas — DISMISSED** [review]

**Review by zancas — APPROVED** [review]

**Pull request merged**